test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/architecture/EmptyManagedKieServersWithSmartRouterAndControllerIntegrationTest.java

```diff
@@ -0,0 +1,287 @@
+
+        smartProcessServicesClient.signal(CONTAINER_ID_ABC, SIGNAL_2_NAME, null);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED), 0, 10)).hasSize(2);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE), 0, 10)).hasSize(4);
+
+        kieServerDEFProcessServicesClient.signal(CONTAINER_ID_DEF, SIGNAL_2_NAME, null);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED), 0, 10)).hasSize(3);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE), 0, 10)).hasSize(3);
+
+        List<TaskSummary> activeTasks = smartTaskServicesClient.findTasks(USER_YODA, 0, 10);
+        assertThat(activeTasks).hasSize(2);
+        assertThat(activeTasks.get(0).getContainerId()).isEqualTo(CONTAINER_ID_GHI);
+        assertThat(activeTasks.get(0).getProcessInstanceId()).isEqualTo(signalProcessInstances.get(4));
+        assertThat(activeTasks.get(1).getContainerId()).isEqualTo(CONTAINER_ID_GHI);
+        assertThat(activeTasks.get(1).getProcessInstanceId()).isEqualTo(signalProcessInstances.get(5));
+
+        smartTaskServicesClient.startTask(CONTAINER_ID_GHI, activeTasks.get(0).getId(), USER_YODA);
+        kieServerGHITaskServicesClient.startTask(CONTAINER_ID_GHI, activeTasks.get(1).getId(), USER_YODA);
+        kieServerGHITaskServicesClient.completeTask(CONTAINER_ID_GHI, activeTasks.get(0).getId(), USER_YODA, Collections.emptyMap());
+        smartTaskServicesClient.completeTask(CONTAINER_ID_GHI, activeTasks.get(1).getId(), USER_YODA, Collections.emptyMap());
+
+        assertThat(smartQueryServicesClient.findProcessesByContainerId(CONTAINER_ID_GHI, 0, 10)).hasSize(2);
+        smartProcessServicesClient.abortProcessInstance(CONTAINER_ID_DEF, signalProcessInstances.get(3));
+
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_COMPLETED, org.kie.api.runtime.process.ProcessInstance.STATE_ABORTED), 0, 10)).hasSize(6);
+        assertThat(smartQueryServicesClient.findProcessInstancesByStatus(Arrays.asList(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE), 0, 10)).hasSize(0);
+        assertThat(smartTaskServicesClient.findTasks(USER_YODA, 0, 10)).hasSize(0);
+    }
+
+    private void waitUntilKieServerLogsContain(KieServerDeployment kieServerDeployment, String logMessage) {
+        for (Instance kieServerInstance : kieServerDeployment.getInstances()) {
+            TimeUtils.wait(Duration.ofSeconds(15), Duration.ofSeconds(1), () -> kieServerInstance.getLogs().contains(logMessage));
+        }
+    }
+
+    private List<Long> createSignalProcesses() {
```

Comment:
This could be made much more concise, e.g

```
private List<Long> createSignalProcesses() {
        // Start 2 instance of SignalUserTask each process on each container
        return Stream.of(CONTAINER_ID_ABC, CONTAINER_ID_ABC, CONTAINER_ID_DEF, CONTAINER_ID_DEF, CONTAINER_ID_GHI, CONTAINER_ID_GHI)
                .map(containerId -> smartProcessServicesClient.startProcess(containerId, SIGNALUSERTASK_PROCESS_ID))
                .collect(toList());
    }
```

The same applies to the remaining 4 instances of this helper method.

Reply:
I change this in unmanaged architectures. In managed mode are processes started throw Smart router and in Kie server.